lib/segment/src/index/hnsw_index/hnsw.rs

```diff
@@ -94,6 +94,25 @@ pub struct HnswIndexOpenArgs<'a> {
     pub stopped: &'a AtomicBool,
 }
 
+fn use_links_compression(existing: bool) -> bool {
```

Comment:
Not sure about `existing` parameter here. How do we handle the case when we create compressed links and reload them? Do we disable enabled compression?

Reply:
Rewritten to make it more clear.

(Read-only) usage of the compression during search is determined based the on-disk file contents. It's not configurable, we can read both formats regardless of the setting.

This setting control when to write it: never/only for new indices/migrate old indices during startup.